Link queue wrapper to log the link queue occupancy (#136)
comunica/comunica-feature-link-traversal: packages/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy/README.md

# Comunica Wrapper Info Occupancy RDF Resolve Hypermedia Links Queue Actor

[![npm version](https://badge.fury.io/js/%40comunica%2Factor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy.svg)](https://www.npmjs.com/package/@comunica/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy)

**Warning: Should not be used in production or in a browser environment**

A comunica Wrapper Info Occupancy RDF Resolve Hypermedia Links Queue Actor.

This module is part of the [Comunica framework](https://github.com/comunica/comunica),
and should only be used by [developers that want to build their own query engine](https://comunica.dev/docs/modify/).

[Click here if you just want to query with Comunica](https://comunica.dev/docs/query/).

This link queue wrapper collects information about the occupancy of the underlying link queues and logs it using a [comunica loggers](https://comunica.dev/docs/query/advanced/logging/) defined by the user. It is preferable to use the [`@comunica/logger-bunyan`](https://github.com/comunica/comunica/tree/master/packages/logger-bunyan).

> The [`comunica-link-queue-parser-rs`](https://github.com/constraintAutomaton/comunica-link-queue-parser-rs) tie-in software can interpret the link queue event and output it as a JSON in the console and/or a file. An example output is presented below.

Reviewer: Could you add a concrete an example usage to the readme of https://github.com/constraintAutomaton/comunica-link-queue-parser-rs ?
To show to people which commands need to be invoked in what order.

Author: I am not sure I understand I thought that the "installation" and "usage" section already explained it.

```bash
cargo build --release
./target/release/comunica-link-queue-parser-rs
```

and the param are all explained.

Reviewer: I mean an actual concrete example, with an actual query.
The current README only shows an example output, but no example input.
The actual commands you need to execute, including comunica link traversal.

Author: It is done

# Comunica Wrapper Info Occupancy RDF Resolve Hypermedia Links Queue Actor

[![npm version](https://badge.fury.io/js/%40comunica%2Factor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy.svg)](https://www.npmjs.com/package/@comunica/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy)

A comunica Wrapper Info Occupancy RDF Resolve Hypermedia Links Queue Actor.

This module is part of the [Comunica framework](https://github.com/comunica/comunica),
and should only be used by [developers that want to build their own query engine](https://comunica.dev/docs/modify/).

[Click here if you just want to query with Comunica](https://comunica.dev/docs/query/).

This link queue wrapper's purpose is to collect information about the occupancy of the link queue in a file.
It should not be used in production or in a browser environment.

> To observe non-standard properties of the link queue the `IOptionalLinkQueueParameters` can be extended and the methods can be modified to process those parameters.

Reviewer: Can you link to https://github.com/constraintAutomaton/comunica-link-queue-parser-rs so readers of this readme can know how they should get and process the output when enabling this actor?

# Comunica Wrapper Info Occupancy RDF Resolve Hypermedia Links Queue Actor

[![npm version](https://badge.fury.io/js/%40comunica%2Factor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy.svg)](https://www.npmjs.com/package/@comunica/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy)

A comunica Wrapper Info Occupancy RDF Resolve Hypermedia Links Queue Actor.

This module is part of the [Comunica framework](https://github.com/comunica/comunica),
and should only be used by [developers that want to build their own query engine](https://comunica.dev/docs/modify/).

[Click here if you just want to query with Comunica](https://comunica.dev/docs/query/).

> The purpose of this wrapper is to collect in a file information about the occupancy of the link queue.

Reviewer 2: ```suggestion
This link queue wrapper's purpose is to collect information about the occupancy of the link queue in a file.
```

It should not be used in production or in a browser environment.

> To observe non-standard properties of the link queue the `IOptionalLinkQueueParameters` can be extended and the methods can be modified in consequence to process those parameters.

Reviewer 2: ```suggestion
To observe non-standard properties of the link queue the `IOptionalLinkQueueParameters` can be extended and the methods can be modified to process those parameters.
```

## Install

```bash
$ yarn add @comunica/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy
```

## Configure

After installing, this package can be added to your engine's configuration as follows:
```text
{
  "@context": [
    ...
    "https://linkedsoftwaredependencies.org/bundles/npm/@comunica/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy/^0.0.0/components/context.jsonld"
  ],
  "actors": [
    ...
    {
      "@id": "urn:comunica:default:rdf-resolve-hypermedia-links-queue/actors#wrapper-info-occupancy",
      "@type": "ActorRdfResolveHypermediaLinksQueueWrapperInfoOccupancy"
    }
  ]
}
```
**It is preferable that the actor is the last called on the bus by the engine by specifying an `beforeActors` adequately.**
### Config Parameters

> * `filePath`: The file where the information will be save.

Reviewer 2: ```suggestion
* `filePath`: The file where the information will be saved.
```

The file name will be extended by an identifier for the query.

> The identifier is either generated by the engine or specify by the user by adding in the
> context the key `@comunica/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupance:query-identifier`
> with an id.

Reviewer 2: ```suggestion
The identifier is either generated by the engine or specified by the user by setting the key `@comunica/actor-rdf-resolve-hypermedia-links-queue-wrapper-info-occupancy:query-identifier`
with an id in the context.
```
Also, where is this query-identifier key specified? Should it not be in the Link Traversal Keys file?

Author: Oups legacy of the previous implementation.
